### PR TITLE
Update to use BOM

### PR DESCRIPTION
**Pull request by bitwiseman**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -218,26 +199,23 @@
...
+            <dependency>
+                <groupId>io.jenkins.tools.bom</groupId>
+                <artifactId>bom-2.150.x</artifactId>
+                <version>5</version>
+                <scope>import</scope>
+                <type>pom</type>
+            </dependency>
             <dependency>
                 <groupId>org.jenkins-ci</groupId>
                 <artifactId>annotation-indexer</artifactId>
```
Comment (jglick) [inline pom.xml]: Note: probably superseded by use of core BOM in 4.x parent. @jtnord could confirm

File: pom.xml [hunk 2]
```diff
@@ -26,7 +26,6 @@
         <hpi.compatibleSinceVersion>2.2.0</hpi.compatibleSinceVersion>
         <java.level>8</java.level>
         <jenkins.version>2.150.3</jenkins.version>
-        <scm-api.version>2.6.3</scm-api.version>
         <hamcrest.version>2.2</hamcrest.version>
         <useBeta>true</useBeta>
         <jcasc.version>1.35</jcasc.version>
```
Comment (jglick) [inline pom.xml]: see https://github.com/jenkinsci/bom/pull/172

File: pom.xml [hunk 3]
```diff
@@ -77,80 +72,67 @@
...
         <!-- Currently just here for interactive testing via hpi:run: -->
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>junit</artifactId>
-            <version>1.26.1</version>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>script-security</artifactId>
```
Comment (jglick) [inline pom.xml]: Could probably be deleted.

File: pom.xml [hunk 4]
```diff
@@ -77,80 +72,67 @@
...
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>scm-api</artifactId>
-            <version>${scm-api.version}</version>
             <classifier>tests</classifier>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>branch-api</artifactId>
```
Comment (jglick) [inline pom.xml]: ditto

File: pom.xml [hunk 5]
```diff
@@ -77,80 +72,67 @@
...
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>pipeline-build-step</artifactId>
-            <version>2.4</version>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-api</artifactId>
```
Comment (jglick) [inline pom.xml]: ditto

File: pom.xml [hunk 6]
```diff
@@ -77,80 +72,67 @@
...
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-job</artifactId>
-            <version>2.31</version>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-scm-step</artifactId>
```
Comment (jglick) [inline pom.xml]: ditto

File: pom.xml [hunk 7]
```diff
@@ -77,80 +72,67 @@
...
-            <version>2.7</version>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-step-api</artifactId>
```
Comment (jglick) [inline pom.xml]: ditto

File: pom.xml [hunk 8]
```diff
@@ -77,80 +72,67 @@
...
-            <version>2.19</version>
             <optional>true</optional>
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-support</artifactId>
```
Comment (jglick) [inline pom.xml]: ditto

File: pom.xml [hunk 9]
```diff
@@ -206,7 +188,6 @@
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-cps-global-lib</artifactId>
-            <version>2.12</version>
             <scope>test</scope>
             <exclusions>
                 <exclusion>
```
Comment (jglick) [inline pom.xml]: Could I guess be deleted if you are pinning a `commons-lang3` version anyway (for some reason).

File: pom.xml (hunk 1 above)
Comment (bitwiseman) [reply pom.xml]: I don't know what this means.

File: pom.xml (hunk 2 above)
Comment (bitwiseman) [reply pom.xml]: So, should we wait on this change until that pr is resolved?

**Comment by bitwiseman** [comment]
> OK but could be further simplified by removing mention of transitive dependencies. (Generally good style to keep them when used during compilation, as for example `structs` for `@Symbol` even though it would be pulled in anyway.)

Is there way to determine this without careful consideration of each dependency? I removed structs and everything still built fine, but it sound like I should put it back?

File: pom.xml (hunk 2 above)
Comment (jglick) [reply pom.xml]: No, just FYI.

File: pom.xml (hunk 1 above)
Comment (jglick) [reply pom.xml]: That you would pick up https://github.com/jenkinsci/jenkins/blob/c376ffe0a41ff4a2100d41ca679a2c7995557dfd/bom/pom.xml#L168-L172 if you updated to the beta parent and got https://github.com/jenkinsci/plugin-pom/pull/269. Can ignore for now.

**Comment by jglick** [comment]
> Is there way to determine this without careful consideration of each dependency?

[Yes](https://github.com/jenkinsci/bom#usage) but I would not worry about it. If and when we strengthen the parent POM to require explicit dependencies for types used during compilation (as for example the Maven harness for NetBeans modules does), you would be informed via build errors of issues. For now this is just stylistic.

File: pom.xml [hunk 10]
```diff
@@ -26,7 +26,6 @@
         <hpi.compatibleSinceVersion>2.2.0</hpi.compatibleSinceVersion>
         <java.level>8</java.level>
         <jenkins.version>2.150.3</jenkins.version>
-        <scm-api.version>2.6.3</scm-api.version>
         <hamcrest.version>2.2</hamcrest.version>
```
Comment (jglick) [inline pom.xml]: It is possible this is obsolete, too. Check whether you can delete all mention of Hamcrest.

File: pom.xml (hunk 1 above)
Comment (jtnord) [reply pom.xml]: should not be required in latest parent pom

**Comment by bitwiseman** [comment]
I was too slow and need to go back and fix the merge.

**Pull request merged**